test/NuGet.Core.Tests/NuGet.PackageManagement.Test/PackageReferenceBasedNuGetPackageManagerTests.cs

```diff
@@ -225,6 +256,102 @@ public async Task PreviewUpdatePackagesAsync_WithMultiTargettedPackageReferenceP
             net50Target.Libraries[0].Version.Should().Be(after.Version);
         }
 
+        [Fact]
+        public async Task PreviewUpdatePackagesAsync_WithMultiTargettedPackageReferenceProject_UpdatesAllConditionalPackages()
+        {
+            // Arrange
+            using SimpleTestPathContext pathContext = new();
+            using var solutionManager = new TestSolutionManager(pathContext);
+            SourceRepositoryProvider sourceRepositoryProvider = TestSourceRepositoryUtility.CreateSourceRepositoryProvider(new PackageSource(pathContext.PackageSource));
+            var settings = Settings.LoadDefaultSettings(solutionManager.SolutionDirectory);
+            var nuGetPackageManager = new NuGetPackageManager(sourceRepositoryProvider, settings, solutionManager, new TestDeleteOnRestartManager());
+
+            string referenceSpec = @"
+                {
+                    ""frameworks"": {
+                        ""net472"": {
+                            ""dependencies"": {
+                                ""A"" : ""1.0.0""
+                            }
+                        },
+                        ""net5.0"": {
+                            ""dependencies"": {
+                                ""B"" : ""1.0.0""
+                            }
+                        }
+                    }
+                }";
+            NuGetProject buildIntegratedProject = CreateBuildIntegratedProjectAndAddToSolutionManager(solutionManager, settings, referenceSpec);
+
+            var packageA = "A";
+            var packageB = "B";
+            var a100 = new PackageIdentity(packageA, new NuGetVersion(1, 0, 0));
+            var a200 = new PackageIdentity(packageA, new NuGetVersion(2, 0, 0));
+            var b100 = new PackageIdentity(packageB, new NuGetVersion(1, 0, 0));
+            var b200 = new PackageIdentity(packageB, new NuGetVersion(2, 0, 0));
+            await SimpleTestPackageUtility.CreateFolderFeedV3Async(pathContext.PackageSource,
+                a100,
+                a200,
+                b100,
+                b200);
+
+            // Main Act
+            var result = (await nuGetPackageManager.PreviewUpdatePackagesAsync(
+                new List<NuGetProject> { buildIntegratedProject },
+                new ResolutionContext(DependencyBehavior.Lowest, false, false, VersionConstraints.None),
+                new TestNuGetProjectContext(),
+                sourceRepositoryProvider.GetRepositories(),
+                sourceRepositoryProvider.GetRepositories(),
```

Comment:
They are combined with an uniqueness check so not really relevant. 

I think this part of the code is not the most interesting part of the changes though. It's fairly standard test setup code.

Reply:
I agree with your feedback. Maybe we could have passed an empty list (Enumerable.Empty<SourceRepository>) because there are no secondary sources in this test.

Reply:
Nearly all of our tests do the same thing, so I don't think it warrants a change. This is what the PM UI would pass in for PR anyways.